File: core/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.java
package com.dbc.code;

public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        int slow = 1, fast = 1;
        while (fast < nums.length) {
            if (nums[fast] != nums[fast - 1]) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        return slow;
    }
}

File: core/RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.py
from typing import List


def removeDuplicates(nums: List[int]) -> int:
    left, right, length = 0, 0, len(nums)
    while right < length:
        while right < length and nums[left] == nums[right]:
            right += 1
        if right < length and right > left + 1:
            nums[left + 1] = nums[right]
        left += 1

    return left
